l1-contracts/contracts/bridge/asset-router/L1AssetRouter.sol

```diff
@@ -0,0 +1,582 @@
+    function _ensureTokenRegisteredWithNTV(address _token) internal returns (bytes32 assetId) {
+        assetId = nativeTokenVault.getAssetId(block.chainid, _token);
+        if (nativeTokenVault.tokenAddress(assetId) == address(0)) {
+            nativeTokenVault.registerToken(_token);
+        }
+    }
+
+    /// @inheritdoc IL1AssetRouter
+    function transferAllowanceToNTV(
+        bytes32 _assetId,
+        uint256 _amount,
+        address _prevMsgSender
+    ) external onlyNativeTokenVault {
+        address l1TokenAddress = INativeTokenVault(address(nativeTokenVault)).tokenAddress(_assetId);
+        if (l1TokenAddress == address(0) || l1TokenAddress == ETH_TOKEN_ADDRESS) {
+            return;
+        }
+        IERC20 l1Token = IERC20(l1TokenAddress);
+
+        // Do the transfer if allowance to Shared bridge is bigger than amount
+        // And if there is not enough allowance for the NTV
+        if (
+            l1Token.allowance(_prevMsgSender, address(this)) >= _amount &&
+            l1Token.allowance(_prevMsgSender, address(nativeTokenVault)) < _amount
+        ) {
+            // slither-disable-next-line arbitrary-send-erc20
+            l1Token.safeTransferFrom(_prevMsgSender, address(this), _amount);
+            l1Token.forceApprove(address(nativeTokenVault), _amount);
+        }
+    }
+
+    /// @dev The request data that is passed to the bridgehub.
+    /// @param _prevMsgSender The `msg.sender` address from the external call that initiated current one.
+    /// @param _assetId The deposited asset ID.
+    /// @param _bridgeMintCalldata The calldata used by remote asset handler to mint tokens for recipient.
+    /// @param _txDataHash The keccak256 hash of 0x01 || abi.encode(bytes32, bytes) to identify deposits.
+    /// @return request The data used by the bridgehub to create L2 transaction request to specific ZK chain.
+    function _requestToBridge(
+        address _prevMsgSender,
+        bytes32 _assetId,
+        bytes memory _bridgeMintCalldata,
+        bytes32 _txDataHash
+    ) internal view virtual returns (L2TransactionRequestTwoBridgesInner memory request) {
+        bytes memory l2TxCalldata = getDepositCalldata(_prevMsgSender, _assetId, _bridgeMintCalldata);
+
+        request = L2TransactionRequestTwoBridgesInner({
+            magicValue: TWO_BRIDGES_MAGIC_VALUE,
+            l2Contract: L2_ASSET_ROUTER_ADDR,
+            l2Calldata: l2TxCalldata,
+            factoryDeps: new bytes[](0),
+            txDataHash: _txDataHash
+        });
+    }
+
+    /// @inheritdoc IL1AssetRouter
+    function getDepositCalldata(
+        address _sender,
+        bytes32 _assetId,
+        bytes memory _assetData
+    ) public view override returns (bytes memory) {
+        // First branch covers the case when asset is not registered with NTV (custom asset handler)
+        // Second branch handles tokens registered with NTV and uses legacy calldata encoding
```

Comment:
I know this has not changed since the original code, but why do we use legacy encoding for the old NTV? I suspect it is to avoid breaking changes during the migration, but it should be documented as it is not clear we would actively promote anything legacy